core/templates/filters/string-utility-filters/truncate-and-capitalize.pipe.ts

```diff
@@ -21,11 +21,16 @@ import { Pipe, PipeTransform } from '@angular/core';
 // Note that this filter does not truncate at the middle of a word.
 @Pipe({name: 'truncateAndCapitalize'})
 export class TruncateAndCapitalizePipe implements PipeTransform {
-  transform(input: string, maxNumberOfCharacters: number): string {
+  // 'maxNumberOfCharacters' can be set to 'null' for returning the entire input
+  // string with the first letter capitalized.
+  transform(input: string, maxNumberOfCharacters: number | null): string {
     if (!input) {
       return input;
     }
-    let words = input.trim().match(/\S+/g);
+    // The following regexp match will only return 'null' on an empty input
+    // string, the condition is already being checked above.
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
`eslint-disable-next-line` why is this needed here?

Reply:
`eslint-disable-next-line` is needed for the following reason:
![Screenshot from 2021-06-16 08-57-49](https://user-images.githubusercontent.com/43031234/122155712-fbc41700-ce80-11eb-9069-ff497b6c8334.png)

and I am not sure by what you mean by why it would be an empty string? and checking for it to be an empty string?

Reply:
I mean why is this comment needed at all, it was not there before? Something like this should work: 
``` typescript
    if (!input && input === '') {
      return input;
    }
    let words = input.trim().match(/\S+/g);

    // Capitalize the first word and add it to the result.
    if (words === null) {
      throw new Error('No words found');
    }
    let result = words[0].charAt(0).toUpperCase() + words[0].slice(1);
```
I have added the condition where we check for it to be an empty string so if null or undefined return the same string and if empty then also return empty. You will have to add a test for error thrown.

Reply:
Yeah, I would agree with @mridul-netizen approach.

Reply:
The `eslint-disable-next-line` was needed because I used a `(!) non-null assertion`  for the line
```typescript
let words = input.trim().match(/\S+/g)!;
```

The comment was needed to explain why `words` can never be `null`. `match()` for this particular regexp will only return `null` on an empty string. Now, since `!input` will be also be `true` for when `input == ''` and this condition was already being checked before the statement where `match()` was called, `words` will simply just never be `null`.

Taking a look at @mridul-netizen approach, the following condition can **never** happen: We cannot even write tests for it. It is unreachable.
```typescript
if (words === null) {
      throw new Error('No words found');
    }
```